awscli/customizations/s3/results.py

```diff
@@ -246,8 +246,11 @@ def _get_ongoing_dict_key(self, result):
                 'Any result using _get_ongoing_dict_key must subclass from '
                 'BaseResult. Provided result is of type: %s' % type(result)
             )
-        return ':'.join(
-            str(el) for el in [result.transfer_type, result.src, result.dest])
+        key_parts = []
+        for result_property in [result.transfer_type, result.src, result.dest]:
+            if result_property is not None:
```

Comment:
I do not have an opinion either way. Pretty confident is not needed given we have a lot of tests for this and we both looked through the code. Plus if you want exact parity, the value would be `"None"` and not empty string, which is weird to rely on. So really the only thing that could break is if we are splitting on `':'` and expecting three elements, which we do not do anywhere in the code.